webapp/src/greenscreen.tsx

```diff
@@ -77,11 +82,18 @@ export class WebCam extends data.Component<WebCamProps, WebCamState> {
     }
 
     componentDidMount() {
-        if (isMediaDevicesSupported())
-            navigator.mediaDevices.enumerateDevices()
-                .then(devices => {
-                    this.setState({ devices: devices.filter(device => device.kind == "videoinput") });
-                })
+        if (isMediaDevicesSupported()) {
+            // first ask for permission from ther user so that
+            // labels are populated in enumerateDevices
+            navigator.mediaDevices.getUserMedia({ audio: false, video: true })
+            .then(() => navigator.mediaDevices.enumerateDevices())
+            .then(devices => {
+                this.setState({ devices: devices.filter(device => device.kind == "videoinput") });
+            }, e => {
+                pxt.debug(`greenscreen: enumerate devices failed`)
+                console.error(e);
+            });
```

Comment:
```suggestion
                .then(() => navigator.mediaDevices.enumerateDevices())
                .then(devices => {
                    this.setState({ devices: devices.filter(device => device.kind == "videoinput") });
                }, e => {
                    pxt.debug(`greenscreen: enumerate devices failed`)
                    console.error(e);
                });
```

nit indentation

Reply:
the default text formatter does not enforce those, might be time to move to prettier who is much more aggressive about indentation,